src/Cake.Svn/Add/SvnAdder.cs

```diff
@@ -26,19 +27,44 @@ public SvnAdder(ICakeEnvironment environment, Func<ISvnClient> clientFactoryMeth
         }
 
         /// <summary>
-        /// Add a file or directory to Subversion.
+        /// Add a directory to Subversion.
         /// </summary>
-        /// <param name="fileOrDirectoryToAdd">The filename or directoryname to add.</param>
+        /// <param name="directory">The directory.</param>
         /// <param name="settings">The settings.</param>
-        /// <returns>True if the command were successful. Otherwise False will be returned.</returns>
-        public bool Add(string fileOrDirectoryToAdd, SvnAddSettings settings)
+        /// <returns>
+        /// <c>true</c> if the command were successful. Otherwise <c>false</c> will be returned.
+        /// </returns>
+        public bool Add(DirectoryPath directory, SvnAddSettings settings)
         {
             settings.NotNull(nameof(settings));
-            fileOrDirectoryToAdd.NotNullOrWhiteSpace(nameof(fileOrDirectoryToAdd));
+            directory.NotNull(nameof(directory));
+
+            var absolutePath = directory.MakeAbsolute(_environment).ToString();
+
+            using (var client = GetClient())
```

Comment:
The using block can be moved to a private methode and reused in both public methods